examples/vision/reptile.py

```diff
@@ -0,0 +1,248 @@
+    old_vars = model.get_weights()
+    # Get a sample from the full dataset.
+    mini_dataset = train_dataset.get_mini_dataset(
+        inner_batch_size, inner_iters, train_shots, classes
+    )
+    for images, labels in mini_dataset:
+        with tf.GradientTape() as tape:
+            logits = model(images)
+            loss = keras.losses.categorical_crossentropy(
+                tf.one_hot(labels, depth=classes), logits
+            )
+        grads = tape.gradient(loss, model.trainable_weights)
+        optimizer.apply_gradients(zip(grads, model.trainable_weights))
+    new_vars = model.get_weights()
+    # Perform SGD for the meta step.
+    for var in range(len(new_vars)):
+        new_vars[var] = old_vars[var] + (
+            (new_vars[var] - old_vars[var]) * cur_meta_step_size
+        )
+    # After the meta-learning step, reload the newly trained weights into the model.
+    model.set_weights(new_vars)
+    # Evaluation loop
+    if meta_iter % eval_interval == 0:
+        accuracies = []
+        for dataset in (train_dataset, test_dataset):
+            # Sample a mini dataset from the full dataset.
+            train_set, test_images, test_labels = dataset.get_mini_dataset(
+                eval_batch_size, eval_iters, shots, classes, split=True
+            )
+            old_vars = model.get_weights()
+            # Train on the samples and get the resulting accuracies.
+            for images, labels in train_set:
+                with tf.GradientTape() as tape:
+                    logits = model(images)
+                    loss = keras.losses.categorical_crossentropy(
+                        tf.one_hot(labels, depth=classes), logits
+                    )
+                grads = tape.gradient(loss, model.trainable_weights)
+                optimizer.apply_gradients(zip(grads, model.trainable_weights))
+            test_preds = model.predict(test_images)
+            test_preds = tf.argmax(test_preds).numpy()
+            num_correct = (test_preds == test_labels).sum()
+            # And reset the weights after getting the evaluation accuracies.
+            model.set_weights(old_vars)
+            accuracies.append(num_correct / classes)
+        training.append(accuracies[0])
+        testing.append(accuracies[1])
+        if meta_iter % 100 == 0:
+            print(
+                "batch %d: train=%f test=%f" % (meta_iter, accuracies[0], accuracies[1])
+            )
+
+"""
+## Visualize Results
+"""
+
+# First, some preprocessing to smooth the training and testing arrays for display.
+window_len = 100
+w = eval("np.hamming(window_len)")
+train_s = np.r_[
+    training[window_len - 1 : 0 : -1], training, training[-1:-window_len:-1]
+]
+test_s = np.r_[testing[window_len - 1 : 0 : -1], testing, testing[-1:-window_len:-1]]
+train_y = np.convolve(w / w.sum(), train_s, mode="valid")
+test_y = np.convolve(w / w.sum(), test_s, mode="valid")
+
+import matplotlib.pyplot as plt
+
+# And display the training accuracies.
```

Comment:
You are only display accuracy -- also display concrete examples. This is what gives the script its meaning.